File: src/389.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    char findTheDifference(string s, string t) {
        char retVal = 0;

        for (char c : s) {
            retVal ^= c;
        }

        for (char c : t) {
            retVal ^= c;
        }

        return retVal;
    }
};

int main(int argc, char** argv) {
    struct subject {
        string s;
        string t;
    };
    vector<subject> testData{{"abcd", "abcde"}, {"", "y"}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input: s = "abcd", t = "abcde"
     *  Output: "e"
     *
     *  Input: s = "", t = "y"
     *  Output: "y"
     */

    Solution cSolution;
    char answer;
    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input: s = \"" << testData[i].s << ", t = \"" << testData[i].t << "\"\n";

        answer = cSolution.findTheDifference(testData[i].s, testData[i].t);
        cout << "Output: \"" << answer << "\"\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}

File: src/389.py
import sys
import logging
from typing import List, Optional
from collections import defaultdict, deque, Counter
from functools import cache, lru_cache
from bisect import bisect_left, bisect_right
from itertools import accumulate
from heapq import heapify, heappush, heappop


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        retVal = 0

        for c in s:
            retVal ^= ord(c)

        for c in t:
            retVal ^= ord(c)

        retVal = chr(retVal)

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for s, t in zip(["abcd", ""], ["abcde", "y"]):
            # /* Example
            #  *  Input: s = "abcd", t = "abcde"
            #  *  Output: "e"
            #  *
            #  *  Input: s = "", t = "y"
            #  *  Output: "y"
            #  */
            logging.info("Input: s = \"%s\", t = \"%s\"", s, t)

            retVal = pSolution.findTheDifference(s, t)
            logging.info("Output: \"%s\"", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__,
                      exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__,
                      exception, exc_info=True)
        pass

    sys.exit(0)
